contracts/sol6/Dao/mock/KyberFeeHandlerWrapper.sol

```diff
@@ -0,0 +1,228 @@
+    /// @param endTokenIndex index of supportedTokens to start iterating from (exclusive)
+    /// @param startKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (inclusive)
+    /// @param endKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (exclusive)
+    /// @return amounts staker reward wei / twei amount claimed from each feeHandler
+    function claimStakerReward(
+        address staker,
+        uint256 epoch,
+        uint256 startTokenIndex,
+        uint256 endTokenIndex,
+        uint256 startKyberFeeHandlerIndex,
+        uint256 endKyberFeeHandlerIndex
+    ) external returns(uint256[] memory amounts) {
+        if (
+            startTokenIndex > endTokenIndex ||
+            startKyberFeeHandlerIndex > endKyberFeeHandlerIndex ||
+            supportedTokens.length == 0
+        ) {
+            // no need to do anything
+            return amounts;
+        }
+
+        uint256 endTokenId = (endTokenIndex >= supportedTokens.length) ?
+            supportedTokens.length : endTokenIndex;
+
+        for (uint256 i = startTokenIndex; i < endTokenId; i++) {
+            KyberFeeHandlerData[] memory kyberFeeHandlerArray = kyberFeeHandlersPerToken[supportedTokens[i]];
+            uint256 endKyberFeeHandlerId = (endKyberFeeHandlerIndex >= kyberFeeHandlerArray.length) ?
+                kyberFeeHandlerArray.length - 1: endKyberFeeHandlerIndex;
+            amounts = new uint256[](endKyberFeeHandlerId - startKyberFeeHandlerIndex + 1);
+
+            // iteration starts from endIndex, differs from claiming reserve rebates and platform wallets
+            for (uint256 j = endKyberFeeHandlerId; j >= startKyberFeeHandlerIndex; j--) {
+                KyberFeeHandlerData memory kyberFeeHandlerData = kyberFeeHandlerArray[j];
+                if (kyberFeeHandlerData.startEpoch < epoch) {
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
+                    break;
+                } else if (kyberFeeHandlerData.startEpoch == epoch) {
+                    amounts[j] = kyberFeeHandlerData.kyberFeeHandler.claimStakerReward(staker, epoch);
+                } else if (j == 0) {
+                    break;
+                } else {
+                    continue;
+                }
+            }
+        }
+    }
+
+    /// @dev claim reabate per reserve wallet. called by any address
+    /// @param rebateWallet the wallet to claim rebates for. Total accumulated rebate sent to this wallet
+    /// @param startTokenIndex index of supportedTokens to start iterating from (inclusive)
+    /// @param endTokenIndex index of supportedTokens to start iterating from (exclusive)
+    /// @param startKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (inclusive)
+    /// @param endKyberFeeHandlerIndex index of feeHandlerArray to start iterating from (exclusive)
```

Comment:
to end iterating?

Reply:
cmd + C and cmd + V too much hahaha.... will fix